data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/MultithreadedProducer.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.producer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSinkConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * * A Multithreaded helper class which helps to produce the records to multiple topics in an
+ * asynchronous way.
+ */
+@SuppressWarnings("deprecation")
+public class MultithreadedProducer implements Runnable {
+    private KafkaProducer<String, String> plainTextProducer = null;
+    private KafkaProducer<String, JsonNode> jsonProducer = null;
+    private KafkaProducer<String, GenericRecord> avroProducer = null;
+    private static final Logger LOG = LoggerFactory.getLogger(MultithreadedProducer.class);
+    private final AtomicBoolean status = new AtomicBoolean(false);
+    private final KafkaSinkConfig sinkConfig;
+
+    private final Record<Event> record;
+
+    private String schemaType;
+    private Properties producerProperties;
+    private PluginMetrics pluginMetrics;
+
+    public MultithreadedProducer(Properties properties,
+                                 KafkaSinkConfig sinkConfig, Record<Event> record,
```

Comment:
I do not think it makes sense to create an instance of a producer for each record.

Reply:
hi @kkondaka , thank you for review comments, we be providing the fix in next commit

Reply:
Please do in this commit.

Reply:
As per the new approach PR #2950 .